gcp/RAILS_DEPLOY_FAQ.md

```diff
@@ -0,0 +1,96 @@
+# Rails deployment Common errors 
+
+## Load both webserver & Sidekiq
+
+It's required to use foreman as the staring point of the application and having a Procfile with web and sidekiq starting commands. More instructions [here](https://cloud.google.com/community/tutorials/appengine-ruby-rails-activejob-sidekiq#deploying_to_app_engine_flexible_environment)
+
+Procfile example 
+```
+web: bundle exec puma -C config/puma.rb
+worker: bundle exec sidekiq -c 5 -v -q mailers
+```
+
+## Set current project
+
+Use this command to change current project `gcloud config set project my-project-id`
+
+## Generate Credentials
+
+If required to generate new `config/master.key` and `config/credentials.yml.enc`, do the following:
+
+```bash
+rm config/credentials.yml.enc
+rm config/master.key
+EDITOR=vim rails credentials:edit 
+```
+
+Copy the secret key base into environment variable `SECRET_KEY_BASE` and content from master.key file into `RAILS_MASTER_KEY`
+
+```yaml
+#Example
+SECRET_KEY_BASE: a041a62984beec5caca6d8807072ead894d9693058333bce7aedca0edaexample101761b40d80e464ff87606eccce4dbc5b0f70c4d34aaa2ff868def09aad42
+RAILS_MASTER_KEY: ec17b39b3be4example7f7038413ddc4
+```
+
+Also add the key into `config/secrets.yml`
+
+```yaml
+production:
+  secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
+```
+
+## App Engine deployment config example
+
+Must be located in the root path of your project. To deploy use `gcloud app deploy app.yaml`
+## app.yaml 
+```yaml
+runtime: ruby
+env: flex # REQUIRED to launch sidekiq & to be able to use private ip for sql
+
+entrypoint: bundle exec foreman start # Must add foreman gem to project
+
+env_variables:
```

Comment:
Is there any way of configuring the environment variables in App Engine's UI? If not, we would need to commit and push this file and it will certainly contain credentials :/

Reply:
There is no way to config env variables on App Engine. 
If you need to commit app.yaml file you can extract variables into another file like this:
```
includes:
  - env_variables.yaml
```

Another option is to configure the CI/CD to retrieve variables from datastore or another keyvault, and set them on the yaml file.

I will add this to the FAQ.